architecture.md

```diff
@@ -0,0 +1,78 @@
+For example, a user could write a system that applies force to an entity by setting a component on that entity, and the physics system would pick that up and react by applying the force to make the entity actually move.
+The modification of entities and components is how server plugins communicate with each other.
+Libraries, however, don’t know about the entities and components.
+
+All of the systems in the backend create a loop, where some systems are proposing changes to entities and their components, and other systems are handling and applying those changes.
+This is called the “Simulation loop”.
+
+Physics, user commands, and scene broadcaster are all systems launched by default in the backend.
+As mentioned earlier, however, even default systems can be added to or removed from this loop.
+For any other functionality, like sensor data processing for example, a new system would have to be added.
+Plugins that generate sensor data and utilize the Ignition Sensors and Ignition Rendering libraries would have to be added to the backend process.
+The visualization of that data, however, is left up to the client side plugins.
+
+## Communication process
+
+Synchronization between processes is performed by scene broadcaster, a server-side Ignition Gazebo plugin that uses Ignition Transport and Ignition Messages libraries to send messages from the server to the client.
+The messages themselves are provided by Ignition Messages, while the framework for creating the publishers and subscribers that exchange messages is from Ignition Transport.
+
+The scene broadcaster system is responsible for getting the state of the world (all of the entities and their component values) from the ever-changing simulation loop running in the back end, packaging that information into a very compact message, and periodically sending it to the frontend to process.
+Omitting this system would mean not being able to visualize data on the frontend, which can be useful for saving computational power.
+
+In addition to state messages sent from the scene broadcaster to the client, the client can also make requests to the server using the user commands system (another backend plugin).
+It also utilizes the Ignition Transport and Ignition Messages libraries.
+The user command system has services that can be requested by the frontend GUI to insert, delete, move, etc., and will send responses back to the GUI acknowledging the receipt and execution of those requests.
+
+
+## Frontend client process
+
+The client-side process, essentially the GUI, also comprises multiple plugins, some loaded by default, others that can be added, and all optional.
+All of the frontend plugins use the Ignition GUI library.
+There are visualization plugins that create the windows of the GUI, add buttons and other interactive features, and a 3D scene plugin that utilizes Ignition Rendering that creates the scene the user sees.
```

Comment:
If we want to add links
https://ignitionrobotics.org/libs/gui
https://ignitionrobotics.org/libs/rendering